datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -3069,6 +3070,8 @@ fn calc_func_dependencies_for_aggregate(
         let group_by_expr_names = group_expr
             .iter()
             .map(|item| item.schema_name().to_string())
+            .collect::<IndexSet<_>>()
+            .into_iter()
```

Comment:
Can we directly remove duplicates from [`group_expr`](https://github.com/jonahgao/datafusion/commit/6eba7f6de31d9e464e5b2fccdf2cc4985f1b8fb6#diff-f69e720234d9da6cb3a4a178d3a5575fcd34f68191335a8c2465a172e8c4e6f1R2953) so that we don't need to perform duplicate removal here again?

Reply:
I played around with trying to make this work and I found it was not easy  -- perhaps we can do it as a follow on PR?